### PR TITLE
fix: compare all chart properies to detect unsaved changes

### PR DESCRIPTION
**Pull request by ZeRego**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1782 #1737

### Description:

- pass saved chart object into explorer provider ( is also going to be useful for future PRs )
- move logic to detect unsaved changes into explorer provider
- unsaved logic now compares all necessary properties 

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Review by rephus — APPROVED** [review]
The problem we had with the unsaved changes was that the chartConfig was getting overwritten with BigNumber Config, not matter what chart we had. But  I guess with your refactoring that should not happen. Code looks ok, but let me QA this to make sure it works as expected

**Comment by rephus** [comment]
Ok, there are a few things that are not working as expected. 

- We should not be able to save from a saved dashboard without chaning anything
Peek 2022-04-25 08-552

Peek 2022-04-25 08-55

- Saving a big number chart, breaks the chart config (you can't set anoter type again) 
Peek 2022-04-25 08-54
(also happening with tables) 
- You can save table chart multiple times (something int he config m ight not match, so it thinks it is always different) 

Peek 2022-04-25 08-53

- When a filter is set, you can save multiple times (button doesn't get disabled) 
- 
Peek 2022-04-25 08-51

Let me know if you need help

**Review by rephus — CHANGES_REQUESTED** [review]
See QA test

**Comment by ZeRego** [comment]
Thanks for testing @rephus 🙏  
- Fixed logic when there are empty properties.
- The issue when changing for table to chart is being tracked under a different ticket https://github.com/lightdash/lightdash/issues/1821 and doesn't seem to be related to this logic.

<a href="https://www.loom.com/share/890c43658e4b4cd6a9cffdcb6845f3b0">
    <p>Lightdash - unsaved changes - Watch Video</p>
    
  </a>

File: packages/common/src/index.ts [hunk 1]
```diff
@@ -1247,3 +1247,34 @@ export function formatRows(
...
+    const newObj: Record<string, any> = {};
+    Object.keys(object).forEach((key) => {
+        if (object[key] === Object(object[key]))
+            newObj[key] = removeEmptyProperties(object[key]);
+        else if (object[key] !== undefined && object[key] !== null)
+            newObj[key] = object[key];
+    });
+    return newObj;
+};
+export const deepEqual = (
```
[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/common/src/index.ts:1262]: I thought about it but didn't want to add an entire library for a single function 🤔 I'm going to bring this up with the team 👍 Not blocking this PR for now

**Pull request merged**